pipeline/matcher_merger/matcher/src/main/scala/weco/pipeline/matcher/MatcherSQSLambda.scala

```diff
@@ -40,14 +41,26 @@ trait MatcherSQSLambda[Config <: ApplicationConfig]
     // each of which contains a set of affected ids
 
     // Next, we must do two things:
-    // 1. notify downstream with all the ids in all the results.
+    // 1. notify downstream with all the MatcherResults.
     // 2. filter out any successful ids from the messagesMap and return the bad messageIds.
     resultsFuture.map {
       results: Iterable[MatcherResult] =>
-        // as a prerequisite for both, flatten sets of ids in MatcherResult into a single set of Strings.
-        val identifiers = results.flatMap(_.allUnderlyingIdentifiers).toSet
-        identifiers.foreach(downstream.notify)
-        findMissingMessages(messagesMap, identifiers)
+        // flatten sets of ids in MatcherResult into a single set of Strings.
+        val initialIdentifiers =
```

Comment:
Not applying to this change set directly, but the extraction of message id and re-mapping of succeeded and failed messages by their contents is adding complexity here that we may be able to abstract.